File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        currentArr = nums[0]
        maxArr = nums[0]
        for i in range(1, len(nums)):
            currentArr = max(nums[i], nums[i] + currentArr)
            maxArr = max(currentArr, maxArr)
        return maxArr

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ans = nums[0];
        int left = 0;
        int currSum = nums[0];
        for (int right = 1; right < nums.length; right++) {
            if (nums[right] > currSum + nums[right]) {
                currSum = nums[right];
            } else {
                currSum += nums[right];
            }
            ans = Math.max(ans, currSum);
        }
        return ans;
    }
}